src/plugins/formulas/formulas.js

```diff
@@ -776,7 +829,12 @@ export class Formulas extends BasePlugin {
    * @returns {*|boolean} If false is returned the action is canceled.
    */
   onBeforeCreateCol(col, amount) {
-    if (!this.engine.isItPossibleToAddColumns(this.sheetId, [this.toPhysicalColumnPosition(col), amount])) {
+    // Using `attempt` because this.sheetId can be null or point to an non-existing sheet
```

Comment:
Same here. Check the conditions or make sure that `this.sheetId` always exist. It should

Reply:
I extended these conditions in https://github.com/handsontable/handsontable/pull/8669/commits/c52a4f85f0966f7c2eeed8e267a374aff3480695.